src/t8_cmesh/t8_cmesh_types.h

```diff
@@ -228,34 +239,41 @@ typedef struct t8_attribute_info
   /**< The size in bytes of the attribute */
 } t8_attribute_info_struct_t;
 
-/* TODO: document, process is a bad naming, since it does not refer to MPI ranks here */
+/**
+ * This structure holds the trees and ghosts of a cmesh.
+ */
 typedef struct t8_cmesh_trees
 {
-  sc_array_t *from_proc;                 /* array of t8_part_tree, one for each process */
-  int *tree_to_proc;                     /* for each tree its process */
-  int *ghost_to_proc;                    /* for each ghost its process */
-  sc_hash_t *ghost_globalid_to_local_id; /* A hash table storing the map
+  sc_array_t *from_proc;                 /**< array of t8_part_tree, one for each process */
+  int *tree_to_proc;                     /**< for each tree its process */
+  int *ghost_to_proc;                    /**< for each ghost its process */
+  sc_hash_t *ghost_globalid_to_local_id; /**< A hash table storing the map
                                                            global_id -> local_id for the ghost trees.
                                                            The local_id is the local ghost id starting at num_local_trees  */
-  sc_mempool_t *global_local_mempool;    /* Memory pool for the entries in the hash table */
+  sc_mempool_t *global_local_mempool;    /**< Memory pool for the entries in the hash table */
 } t8_cmesh_trees_struct_t;
 
-/* TODO: document */
+/**
+ * t8_part_tree
+ * This structure holds the data of a trees in a partition.
+ */
 typedef struct t8_part_tree
 {
-  char *first_tree;           /* Stores the trees, the ghosts and the attributes.
+  char *first_tree;           /**< Stores the trees, the ghosts and the attributes.
                                            The last 2*sizeof(t8_locidx) bytes store num_trees and num_ghosts */
-  t8_locidx_t first_tree_id;  /* local tree_id of the first tree. -1 if num_trees = 0 */
-  t8_locidx_t first_ghost_id; /* TODO: document. -1 if num_ghost=0, 0 for the first part, (not num_local_trees!)
+  t8_locidx_t first_tree_id;  /**< local tree_id of the first tree. -1 if num_trees = 0 */
+  t8_locidx_t first_ghost_id; /**< TODO: document. -1 if num_ghost=0, 0 for the first part, (not num_local_trees!)
                                            0 <= first_ghost_id < num_ghosts */
-  t8_locidx_t num_trees;
-  t8_locidx_t num_ghosts;
+  t8_locidx_t num_trees;      /**< The number of trees */
+  t8_locidx_t num_ghosts;     /**< The number of ghosts */
 } t8_part_tree_struct_t;
 
 /* TODO: Extend this structure with meaningful entries.
  *       Maybe the number of shipped trees per process is useful?
  */
-/** This struct is used to profile cmesh algorithms.
+/** 
+ * t8_cprofile_struct_t
```

Comment:
do we add the name or not? I think not because it is easier, we do not have to change it if the class name changes and it clutters the outpot when hovering over a variable with the mouse